MVP-java/app/src/main/java/com/base/mvp/core/di/module/ApplicationModule.java

```diff
@@ -43,28 +48,46 @@ Application provideApplication() {
         return application;
     }
 
+    @Provides
+    @DatabaseInfo
+    String provideDatabaseName() {
+        return AppConstants.DB_NAME;
+    }
+
+    @Provides
+    @ApiInfo
+    String provideApiKey() {
+        return BuildConfig.API_KEY;
+    }
+
+    @Provides
+    @PreferenceInfo
+    String providePreferenceName() {
+        return AppConstants.PREF_NAME;
+    }
+
     @Provides
     @Singleton
-    DataManager provideDataManager(DataManagerImplement dataManagerImpl) {
-        return dataManagerImpl;
+    DataManager provideDataManager(DataManagerImplement dataManagerImplement) {
+        return dataManagerImplement;
```

Comment:
how different between dataManagerImplement vs dataManagerImpl??? why can you change?

Reply:
i change it to clear code